package/src/KetchServiceProvider/KetchServiceProvider.tsx

```diff
@@ -205,8 +205,9 @@ export const KetchServiceProvider: React.FC<KetchServiceProviderParams> = ({
         break;
 
       case EventName.consent:
-        consent.current = JSON.parse(data.data || {}) as Consent;
-        parameters.onConsentUpdated?.(data.data);
+        const serializedData = JSON.parse(data.data || {}) as Consent;
```

Comment:
I'd renamed `serializedData` to something like `consent`, as it is actually de-serialized data